appinventor/components/src/com/google/appinventor/components/runtime/Ev3Motors.java

```diff
@@ -0,0 +1,664 @@
+    String functionName = "RotateInDistance";
+    int tachoCounts = (int) (distance * 360.0 / wheelDiameter / Math.PI);
+
+    try {
+      if (regulationEnabled)
+        outputStepSpeed(functionName, 0, motorPortBitField, power, 0, tachoCounts, 0, useBrake);
+      else
+        outputStepPower(functionName, 0, motorPortBitField, power, 0, tachoCounts, 0, useBrake);
+    } catch (IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, functionName, ErrorMessages.ERROR_EV3_ILLEGAL_ARGUMENT, functionName);
+    }
+  }
+
+  /**
+   * Start to rotate the motors at the same speed.
+   */
+  @SimpleFunction(description = "Start to rotate the motors at the same speed.")
+  public void RotateSyncIndefinitely(int power, int turnRatio) {
+    String functionName = "RotateSyncIndefinitely";
+
+    try {
+      outputStepSync(functionName, 0, motorPortBitField, power, turnRatio, 0, true);
+    } catch (IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, functionName, ErrorMessages.ERROR_EV3_ILLEGAL_ARGUMENT, functionName);
+    }
+  }
+
+  /**
+   * Rotate the motors at the same speed in a period of time.
+   */
+  @SimpleFunction(description = "Rotate the motors at the same speed in a period of time.")
+  public void RotateSyncInDuration(int power, int milliseconds, int turnRatio, boolean useBrake) {
+    String functionName = "RotateSyncInDuration";
+
+    try {
+      outputTimeSync(functionName, 0, motorPortBitField, power, turnRatio, milliseconds, useBrake);
+    } catch (IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, functionName, ErrorMessages.ERROR_EV3_ILLEGAL_ARGUMENT, functionName);
+    }
+  }
+
+  /**
+   * Rotate the motors at the same speed in a number of tacho counts.
+   */
+  @SimpleFunction(description = "Rotate the motors at the same speed in a number of tacho counts.")
+  public void RotateSyncInTachoCounts(int power, int tachoCounts, int turnRatio, boolean useBrake) {
+    String functionName = "RotateSyncInTachoCounts";
+
+    try {
+      outputStepSync(functionName, 0, motorPortBitField, power, turnRatio, tachoCounts, useBrake);
+    } catch (IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, functionName, ErrorMessages.ERROR_EV3_ILLEGAL_ARGUMENT, functionName);
+    }
+  }
+
+  /**
+   * Stop the motors of the robot.
+   */
+  @SimpleFunction(description = "Stop the motors of the robot.")
+  public void Stop(boolean useBrake) {
+    String functionName = "Stop";
+    try {
+      stopOutput(functionName, 0, motorPortBitField, useBrake); // assume layer = 0
+    } catch (IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, functionName, ErrorMessages.ERROR_EV3_ILLEGAL_ARGUMENT, functionName);
+    }
+  }
+
+  /**
+   * Make motors rotate forwardly.
+   */
+  @SimpleFunction(description = "Make motors rotate forwardly.")
+  public void SetPolarityForward() {
```

Comment:
Changing "Polarity" to "Direction" might make more sense to users. Also, it'd be nice if this could be set from the designer (and also blocks).